public/index.html

```diff
@@ -1,30 +1,35 @@
 <!doctype html>
 <html lang="en">
-  <head>
-    <base href="/" />
-    <meta charset="utf-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <link rel="shortcut icon" href="<%=htmlWebpackPlugin.options.FAVICON_URL%>" type="image/x-icon" />
 
-    <% _.forEach(htmlWebpackPlugin.options.preconnect, function(preconnectURL) { %>
-      <!-- Include preconnect resource hint -->
-      <link rel="preconnect" href="<%= preconnectURL %>" />
-      <!-- Include preconnect resource hint with crossorigin, should the resource use CORS  -->
-      <link rel="preconnect" href="<%= preconnectURL %>" crossorigin />
-      <!-- Fallback to dns-prefetch resource hint if preconnect is not supported by browser -->
-      <link rel="dns-prefetch" href="<%= preconnectURL %>" />
+<head>
+  <base href="/" />
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link rel="shortcut icon" href="<%=htmlWebpackPlugin.options.FAVICON_URL%>" type="image/x-icon" />
+
+  <% _.forEach(htmlWebpackPlugin.options.preconnect, function(preconnectURL) { %>
+    <!-- Include preconnect resource hint -->
+    <link rel="preconnect" href="<%= preconnectURL %>" />
+    <!-- Include preconnect resource hint with crossorigin, should the resource use CORS  -->
+    <link rel="preconnect" href="<%= preconnectURL %>" crossorigin />
+    <!-- Fallback to dns-prefetch resource hint if preconnect is not supported by browser -->
+    <link rel="dns-prefetch" href="<%= preconnectURL %>" />
     <% }); %>
 
-    <% if (htmlWebpackPlugin.options.NODE_ENV==='production' && htmlWebpackPlugin.options.OPTIMIZELY_PROJECT_ID &&
-            htmlWebpackPlugin.options.DEPLOYMENT_ENV==='production') { %>
-      <script src="https://www.edx.org/optimizelyjs/<%= htmlWebpackPlugin.options.OPTIMIZELY_PROJECT_ID %>.js"></script>
-    <% } %>
+      <% if (htmlWebpackPlugin.options.NODE_ENV==='production' && htmlWebpackPlugin.options.OPTIMIZELY_PROJECT_ID &&
+        htmlWebpackPlugin.options.DEPLOYMENT_ENV==='production' ) { %>
+        <script src="https://www.edx.org/optimizelyjs/<%= htmlWebpackPlugin.options.OPTIMIZELY_PROJECT_ID %>.js"
+          async="async"></script>
```

Comment:
[sanity check] Would we want `async` or `defer`?

> If the async attribute is set, the script is downloaded in parallel to parsing the page, and executed as soon as it is available. **The parsing of the page is interrupted once the script is downloaded completely, and then the script is executed, before the parsing of the rest of the page continues.**

[sanity check x2] Is there any risk that we'd be trying to execute Optimizely related code paths _before_ it's fully loaded after these changes? E.g., right now, I believe the earliest call to Optimizely is within `EnterprisePage` where `pushUserCustomerAttributes` is called. However, what if this call was actually done within a route loader? With the Optimizely script downloaded async/deferred, can we still guarantee `window.optimizely` will exist when we need it to?

Reply:
Sanity check 1: I considered both but the safer option of the two was `async` imo. By allowing the execution of the script as soon as its available should reduce the chances of the resources being unavailable on page load. `defer` is an option as well and might be the more performant option but we run the risk of the resources that not being available...

Sanity check 2: ...that being said, I ran a couple of instances on localhost stage with the request blocked and the page did not error out. By parsing the code they all seem to be gated by a check for optimizely before they execute within [this utility file](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/33652b9dd38631d0e07573f6fc833b406fa64f08/src/utils/optimizely.js#L9). Given the appropriate checks we COULD use `defer` as opposed to `async`.

Reply:
[update, follow-up from Slack] We decided to defer on this approach for now given the uncertainty around timing when making Optimizely a non-render-blocking resource (i.e., how do we delay execution of Optimizely helper functions until we know for sure `window.optimizely` has finished loading?). This uncertainty could lead to unintended loss of data pushed up to Optimizely (user identify calls, events, etc.).

Helpful resource: https://support.optimizely.com/hc/en-us/articles/4410289790221-Install-the-snippet-as-a-non-blocking-resource